rpc/internal/collector.service.go

```diff
@@ -164,7 +176,7 @@ func (sc *streamsCoordinator) addStream(name string, registry *poplar.RecorderRe
 	defer group.mu.Unlock()
 
 	id := utility.RandomString()
-	group.streams[id] = make(chan error)
+	group.streams[id] = &stream{buffer: make(chan *events.Performance, 1000)} // TODO: figure out good buffer size
```

Comment:
Are we assuming that sending to the stream will never block because we've made the channel sufficiently large?

Reply:
Or just won't block as much?

Reply:
If we don't exceed this length, this code is fine. I guess my concern was that if we're assuming it can exceed this length, `addEvent()` can deadlock if the stream's buffer becomes full. Since `addEvent()` it takes the mutex lock and then tries putting the event in `stream.buffer`, the insertion will block waiting for an item to be dequeued from the channel (which can't be done because it's taken the lock).

Reply:
Oh right, good catch. Then I should fail if the buffer is full rather than block.

Reply:
Ok, I changed it so that `addEvent` will error if the buffer is full. Raiden is going to run some tests with this code to gauge a reasonable size for the event buffer. I actually think it is better to fail on the event buffer; the whole point is that streams should be adding events at similar cadences. Thus, if your buffer is full problem means you should change your code.